creme/activities/views/calendar.py

```diff
@@ -395,6 +398,14 @@ class ActivityDatesSetting(generic.base.EntityRelatedMixin, generic.CheckedView)
     def get_related_entity_id(self):
         return get_from_POST_or_404(self.request.POST, key=self.activity_id_arg, cast=int)
 
+    def get_calendar_config(self, activity, user):
+        owner = get_user_model().filter(pk=activity.user_id).first()
+
+        if owner is None:
+            owner = user
```

Comment:
If I understand sometimes the owner of the activity has not been assigned yet
1. A simpler code would be `owner = activity.user if activity.user_id else user`
2. I do not understand the logic. I presume the hours constraints concern the participants, & so the activity should be checked with the config items of the users participants; the owner of the activity & the logged user seem not relevant.